### PR TITLE
Bug hunting/fix matcher update

### PR DESCRIPTION
**Pull request by thallesfreitaszup**
## Issue Description

Not updating matcher when undeploy finish

### CODE REVIEW

**Review by barbararochazup — APPROVED** [review]

File: moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt [hunk 1]
```diff
@@ -85,6 +91,27 @@ open class UndeployInteractorImpl @Inject constructor(
         }
     }
 
+    private fun updateStatusInCircleMatcher(circle: Circle, workspace: Workspace) {
```
Comment (leandrorodrigueszup) [inline moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: I didn't get this rule. When an undeploy is done in a circle, will this circle be inactive?

Comment (angelicalimazup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: You did =)
that is the rule.

**Review by angelicalimazup — APPROVED** [review]

File: moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt (hunk 1 above)
Comment (thallesfreitaszup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: yes, operator has no undeploy callback, in line 54 the deployment status is updated to UNDEPLOYED on database.

Comment (leandrorodrigueszup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: Ok. Question: undeploy means removing *everything* from the circle?

Comment (thallesfreitaszup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: yes, all components and routes of circle will be removed from cluster

Comment (leandrorodrigueszup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: Even though the deploy was incremental, will the undeploy remove all?

Comment (thallesfreitaszup) [reply moove/application/src/main/kotlin/io/charlescd/moove/application/deployment/impl/UndeployInteractorImpl.kt:94]: yes, today we only remove all on undeploy and we have a incremental deploy only in default circle

**Review by leandrorodrigueszup — APPROVED** [review]

**Pull request merged**